### PR TITLE
140 have a way to document the parameters

### PR DESCRIPTION
**Pull request by tjadevries**

### CODE REVIEW

File: src/application_settings/_private/file_operations.py [hunk 1]
```diff
@@ -89,31 +82,25 @@ def save(path: Path, data: dict[str, Any]) -> None:
...
-        data_stored = tomllib.load(fptr)
-    return data_stored
-
-
-def _load_json(path: Path) -> dict[str, Any]:
-    data_stored: dict[str, Any] = {}
-    with path.open(mode="r") as fptr:
-        data_stored = json.load(fptr)
-    return data_stored
+def _get_loader(ext: str) -> LoaderOpt:
```
Comment (JJdeVries) [inline src/application_settings/_private/file_operations.py:90]: I think it's a bit cleaner if the get methods would get a `Path` type and split of the extension (the `path.suffix[1:].lower`) themselves.

File: src/application_settings/_private/toml_file_operations.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# import sys
+from pathlib import Path
+from typing import Any
+
+import tomlkit
+
+# if sys.version_info >= (3, 11):
+#     import tomllib
+# else:
+#     import tomli as tomllib
```
Comment (JJdeVries) [inline src/application_settings/_private/toml_file_operations.py:11]: commented stuff can be removed i think

File: src/application_settings/type_notation_helper.py [hunk 3]
```diff
@@ -1,17 +1,23 @@
 # pylint: disable=consider-alternative-union-syntax, useless-suppression
 """Defines type aliases that handle notational differences between python versions."""
 import sys
+from collections.abc import Callable
```
Comment (JJdeVries) [inline src/application_settings/type_notation_helper.py:4]: Why not `typing.Callable`? I think that one is actually included now (through the collections.abc import)

File: src/application_settings/_private/json_file_operations.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+"""Funtions for storing dicts to and loading dicts from file."""
```
Comment (JJdeVries) [inline src/application_settings/_private/json_file_operations.py:1]: small typo `Functions`

**Review by JJdeVries — CHANGES_REQUESTED** [review]

File: src/application_settings/type_notation_helper.py (hunk 3 above)
Comment (tjadevries) [reply src/application_settings/type_notation_helper.py:4]: It's the other way around: typing.Callable is deprecated (https://docs.python.org/3/library/typing.html#typing.Callable) and that just imports collections.abc.Callable

**Pull request merged**